x-pack/plugins/security_solution/public/detections/containers/detection_engine/alerts/use_host_isolation_status.tsx

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isEmpty } from 'lodash';
+import { useEffect, useState } from 'react';
+import { Maybe } from '../../../../../../observability/common/typings';
+import { useAppToasts } from '../../../../common/hooks/use_app_toasts';
+import { getHostMetadata } from './api';
+import { ISOLATION_STATUS_FAILURE } from './translations';
+
+interface HostIsolationStatusResponse {
+  loading: boolean;
+  isIsolated: Maybe<boolean>;
+}
+
+/*
+ * Retrieves the current isolation status of a host */
+export const useHostIsolationStatus = ({
+  agentId,
+}: {
+  agentId: string;
+}): HostIsolationStatusResponse => {
+  const [isIsolated, setIsIsolated] = useState<Maybe<boolean>>();
+  const [loading, setLoading] = useState(false);
+
+  const { addError } = useAppToasts();
+
+  useEffect(() => {
+    // isMounted tracks if a component is mounted before changing state
+    let isMounted = true;
+    setLoading(true);
+    const fetchData = async () => {
+      try {
+        const metadataResponse = await getHostMetadata({ agentId });
+        if (isMounted) {
+          setIsIsolated(Boolean(metadataResponse.metadata.Endpoint.configuration.isolation));
+        }
+      } catch (error) {
+        addError(error.message, { title: ISOLATION_STATUS_FAILURE });
+      }
+    };
+    if (isMounted) {
+      setLoading(false);
```

Comment:
This does not look right to me. this will always be executed because `isMounted` will always be `true` at this stage. 

I'm thinking this `if(){}` statement is meant to be inside of the `fetchData()` above on like 45 (after the `catch()`?

Reply:
yup ur right! good catch